admin-dev/themes/new-theme/js/components/tinymce-editor.js

```diff
@@ -0,0 +1,217 @@
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+const $ = window.$;
+
+/**
+ * This class init TinyMCE instances in the back-office. It is wildly inspired by
+ * the scripts from js/admin And it actually loads TinyMCE from the js/tony_mce
+ * folder along with its modules. One improvement could be to install TinyMCE via
+ * npm and fully integrate in the back-office theme.
+ */
+class TinyMCEEditor {
+  constructor(options) {
+    options = options || {};
+    this.tinyMCELoaded = false;
+    this.setupTinyMCE(options);
+  }
+
+  /**
+   * Initial setup which checks if the tinyMCE library is already loaded.
+   *
+   * @param config
+   */
+  setupTinyMCE(config) {
+    if (typeof tinyMCE === 'undefined') {
+      this.loadAndInitTinyMCE(config);
+    } else {
+      this.initTinyMCE(config);
+    }
+  }
+
+  /**
+   * Prepare the config and init all TinyMCE editors
+   *
+   * @param config
+   */
+  initTinyMCE(config) {
+    config = Object.assign({
+      selector: ".rte",
+      plugins: "align colorpicker link image filemanager table media placeholder advlist code table autoresize",
+      browser_spellcheck: true,
+      toolbar1: "code,colorpicker,bold,italic,underline,strikethrough,blockquote,link,align,bullist,numlist,table,image,media,formatselect",
+      toolbar2: "",
+      external_filemanager_path: baseAdminDir + "filemanager/",
```

Comment:
Is `baseAdminDir` a global JS variable ? Shouldn't we put it as a constructor argument ?

Reply:
Yes it's a global variable declared in https://github.com/PrestaShop/PrestaShop/blob/develop/admin-dev/header.inc.php#L57

Reply:
I will inject it in the constructor indeed

Reply:
Ok I added the possibility to set `baseAdminUrl` and `langIsRtl` config in `TinyMCEEditor`
I prefered to manage the fallback in `TinyMCEEditor` constructor, so that it remains easy to integrate (you only need to add `new TinyMCEEditor();` to use it).